Packages/vcs/Lib/vcsvtk/isolinepipeline.py

```diff
@@ -0,0 +1,224 @@
+                else:
+                    colorOverrides = [None] * numLevels
+
+                # Gather text objects
+                if self._gm.text:
+                    texts = self._gm.text
+                    while len(texts) < numLevels:
+                        texts.append(texts[-1])
+                else:
+                    texts = [None] * numLevels
+
+                tprops = vtk.vtkTextPropertyCollection()
+                for tc, colorOverride in zip(texts, colorOverrides):
+                    if tc is None:
+                        tt = "default"
+                        to = "default"
+                    elif vcs.queries.istextcombined(tc):
+                        tt, to = tuple(tc.name.split(":::"))
+                    elif vcs.queries.istexttable(tc):
+                        tt = tc.name
+                        to = "default"
+                    elif vcs.queries.istextorientation(tc):
+                        to = tc.name
+                        tt = "default"
+                    if colorOverride is not None:
+                        tt = vcs.createtexttable(None, tt)
+                        tt.color = colorOverride
+                        tt = tt.name
+                    tprop = vtk.vtkTextProperty()
+                    vcs2vtk.prepTextProperty(
+                          tprop, self._context.renWin.GetSize(), to, tt)
+                    tprops.AddItem(tprop)
+                    if colorOverride is not None:
+                        del(vcs.elements["texttable"][tt])
+
+                mapper.SetTextProperties(tprops)
+            else:
+                # No text properties specified. Use the default:
+                tprop = vtk.vtkTextProperty()
+                vcs2vtk.prepTextProperty(tprop, self._context.renWin.GetSize())
+                mapper.SetTextProperty(tprop)
+
+        stripper = vtk.vtkStripper()
+        stripper.SetInputConnection(cot.GetOutputPort())
+        mapper.SetInputConnection(stripper.GetOutputPort())
+        # TODO remove update, make pipeline
+        stripper.Update()
+        mappers.append([mapper,])
+        self._resultDict["vtk_backend_contours"] = [cot,]
+
+        if self._maskedDataMapper is not None:
+            mappers.insert(0, self._maskedDataMapper)
+
+        x1, x2, y1, y2 = vcs.utils.getworldcoordinates(self._gm,
+                                                       self._data1.getAxis(-1),
+                                                       self._data1.getAxis(-2))
+
+        # And now we need actors to actually render this thing
+        actors = []
+        for mapper in mappers:
+            act = vtk.vtkActor()
+            # TODO Charles -- Mapper is always a list, what is
+            # this supposed to mean?
+            if isinstance(mapper, list):
+                act.SetMapper(mapper[0])
+            else:
+                mapper.Update()
+                act.SetMapper(mapper)
```

Comment:
@doutriaux1 For isolines, all entries in `mappers` are single-element lists with a single mapper. Is there a reason for the extra level of list nesting?